apm-agent-api/src/main/java/co/elastic/apm/api/Transaction.java

```diff
@@ -67,6 +67,18 @@
     @Nonnull
     Transaction setFrameworkName(String frameworkName);
 
+    /**
+     * Sets the service name and version for this transaction and its child spans.
+     * <p>
+     * NOTE: If this method is called after child spans are already created,
+     * they may have the wrong service name and version.
+     * </p>
+     *
+     * @param serviceInfo the service name and version
+     */
+    @Nonnull
+    Transaction setServiceInfo(ServiceInfo serviceInfo);
```

Comment:
Set the service name and method as `String` args (through the same API method)

Reply:
Done